RenderMyCanvas/src/DrawBoard/DrawBoard.cpp

```diff
@@ -1,92 +1,207 @@
-                ClearTemporaryPrimitive();
-                AddPrimitive(CreatePrimitiveFromPoints(m_CurrentPoints));
-                m_CurrentPoints.clear();
-            }
-        }
-        else if (action == 0) {
-            m_IsDrawing = false;
-			m_CurrentPoints.push_back(currentPos);
-            if (m_CurrentPoints.size() >= RequiredPointsForCurrentShape()) {
-                SetTemporaryPrimitive(CreateTemporaryPrimitive(m_CurrentPoints));
-            }
-			m_CurrentPoints.pop_back();
-        }
+DrawBoard::DrawBoard() {
+  m_uv0 = ImVec2(0, 0);
+  m_uv1 = ImVec2(1, 1);
+}
+
+void DrawBoard::OnMouseEvent(int action, float x, float y) {
+  glm::vec2 currentPos(x, y);
+
+  if (action == 1 && m_IsDrawing == false) {
+    m_IsDrawing = true;
+    m_CurrentPoints.push_back(currentPos);
+    if (m_CurrentPoints.size() >= RequiredPointsForCurrentShape()) {
+      ClearTemporaryPrimitive();
+      AddPrimitive(CreatePrimitiveFromPoints(m_CurrentPoints));
+      m_CurrentPoints.clear();
+    }
+  } else if (action == 0) {
+    m_IsDrawing = false;
+    m_CurrentPoints.push_back(currentPos);
+    if (m_CurrentPoints.size() >= RequiredPointsForCurrentShape()) {
+      SetTemporaryPrimitive(CreateTemporaryPrimitive(m_CurrentPoints));
     }
+    m_CurrentPoints.pop_back();
+  }
+}
+
+void DrawBoard::Render(const Scene &scene, const Camera &camera) {
+  std::memset(m_ImageData, 0,
+              m_FinalImage->GetWidth() * m_FinalImage->GetHeight() *
+                  sizeof(uint32_t));
+
+  for (auto &primitive : m_Primitives) {
+    if (primitive)
+      primitive->Draw(m_ImageData, m_FinalImage->GetWidth(),
+                      m_FinalImage->GetHeight());
+  }
+
+  if (m_TemporaryPrimitive)
+    m_TemporaryPrimitive->Draw(m_ImageData, m_FinalImage->GetWidth(),
+                                 m_FinalImage->GetHeight());
 
-    void DrawBoard::Render(const Scene& scene, const Camera& camera) {
-        std::memset(m_ImageData, 0, m_FinalImage->GetWidth() * m_FinalImage->GetHeight() * sizeof(uint32_t));
+  m_FinalImage->SetData(m_ImageData);
+}
 
-        for (auto& primitive : m_Primitives) {
-			if (primitive)
-            primitive->Draw(m_ImageData, m_FinalImage->GetWidth(), m_FinalImage->GetHeight());
-        }
+void DrawBoard::AddPrimitive(std::shared_ptr<Primitive> primitive) {
+  m_Primitives.push_back(primitive);
+}
 
-        if (m_TemporaryPrimitive)
-            m_TemporaryPrimitive->Draw(m_ImageData, m_FinalImage->GetWidth(), m_FinalImage->GetHeight());
+void DrawBoard::SetTemporaryPrimitive(std::shared_ptr<Primitive> primitive) {
+  m_TemporaryPrimitive = primitive;
+}
 
-        m_FinalImage->SetData(m_ImageData);
+void DrawBoard::ClearTemporaryPrimitive() { m_TemporaryPrimitive.reset(); }
+
+std::shared_ptr<Primitive> DrawBoard::CreatePrimitiveFromPoints(const std::vector<glm::vec2>& points) {
+    switch (m_CurrentDrawingMode) {
+    case DrawingMode::Line:
+        return std::make_shared<Line>(glm::vec3(points[0], 0.0f),
```

Comment:
Only pass the point vector to Primitive Constructor. Do not do math calculation here.